lib/topological_inventory/openshift/operations/core/service_catalog_client.rb

```diff
@@ -1,64 +1,58 @@
 require "topological_inventory/openshift/operations/core/authentication_retriever"
-require "topological_inventory/openshift/operations/core/service_plan_client"
 require "topological_inventory-api-client"
 
 module TopologicalInventory
   module Openshift
     module Operations
       module Core
         class ServiceCatalogClient
+          attr_accessor :default_endpoint, :authentication, :connection_manager
+
           def initialize(source_id)
             api_client = TopologicalInventoryApiClient::DefaultApi.new
 
             all_source_endpoints = api_client.list_source_endpoints(source_id)
-            @default_endpoint = all_source_endpoints.data.find { |endpoint| endpoint.default }
+            self.default_endpoint = all_source_endpoints.data.find { |endpoint| endpoint.default }
 
-            authentication_id = api_client.list_endpoint_authentications(@default_endpoint.id.to_s).data.first&.id
+            authentication_id = api_client.list_endpoint_authentications(default_endpoint.id.to_s).data.first&.id
+            self.authentication = AuthenticationRetriever.new(authentication_id).process
 
-            @authentication = AuthenticationRetriever.new(authentication_id).process
+            self.connection_manager = TopologicalInventory::Openshift::Connection.new
           end
 
           def order_service_plan(plan_name, service_offering_name, additional_parameters)
-            payload = ServicePlanClient.new.build_payload(plan_name, service_offering_name, additional_parameters)
-            response = external_request(:post, order_service_plan_url, payload)
-            JSON.parse(response.body)
-          end
-
-          private
+            connection = connection_manager.connect(
+              "servicecatalog", host: default_endpoint.host, token: authentication.password, verify_ssl: verify_ssl_mode)
 
-          def order_service_plan_url
-            base_url_path = URI::Generic.build(
-              :scheme => @default_endpoint.scheme,
-              :host   => @default_endpoint.host,
-              :port   => @default_endpoint.port,
-              :path   => @default_endpoint.path
-            ).to_s
-            URI.join(base_url_path, "apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances").to_s
+            payload = build_payload(plan_name, service_offering_name, additional_parameters)
+            connection.create_service_instance(payload)
           end
 
-          def external_request(method, url, payload, headers = generic_headers)
-            request_options = {
-              :method     => method,
-              :url        => url,
-              :headers    => headers,
-              :verify_ssl => verify_ssl_mode,
-              :payload    => payload
-            }
+          private
 
-            RestClient::Request.new(request_options).execute
-          end
+          def build_payload(service_plan_name, service_offering_name, order_parameters)
```

Comment:
build_payload was an instance method in ServicePlanClient but didn't depend on any instance vars so I just moved it in here so we could delete another class